tests/benchmark/bm_datasets.py

```diff
@@ -109,26 +119,28 @@ def measure_recall_per_second(hnsw_index, dataset, num_queries, k, ef_runtime):
     print("HNSW query per seconds: ", num_queries/hnsw_total_time)
 
 
-def run_benchmark(dataset_name, ef_construction, M, ef_values, k=10):
+def run_benchmark(dataset_name, is_multi, data_type, ef_construction, M, ef_values, k=10):
+    if data_type != VecSimType_FLOAT32 and data_type != VecSimType_FLOAT64:
+        raise TypeError("run_benchmark: This datatype is not supported")
+
     print("\nRunning benchmark for:", dataset_name)
     dataset = get_data_set(dataset_name)
     index_file_name = os.path.join('data', '%s-M=%s-ef=%s.hnsw' % (dataset_name, M, ef_construction))
-    hnsw_index = load_or_create_hnsw_index(dataset, index_file_name, ef_construction, M)
+    hnsw_index = load_or_create_hnsw_index(dataset, index_file_name, is_multi, data_type, ef_construction, M)
    
     for ef_runtime in ef_values:
-        measure_recall_per_second(hnsw_index, dataset, num_queries=1000, k=k, ef_runtime=ef_runtime)
-
+        measure_recall_per_second(hnsw_index, dataset, is_multi, data_type, num_queries=1000, k=k, ef_runtime=ef_runtime)
 
 if __name__ == "__main__":
     DATASETS = ['glove-25-angular', 'glove-50-angular', 'glove-100-angular', 'glove-200-angular',
                 'mnist-784-euclidean', 'sift-128-euclidean']
-    # for every dataset the params are: (ef_construction, M, ef_runtime).
-    dataset_params = {'glove-25-angular': (100, 16, [50, 100, 200]),
-                      'glove-50-angular': (150, 24, [100, 200, 300]),
-                      'glove-100-angular': (250, 36, [150, 300, 500]),
-                      'glove-200-angular': (350, 48, [200, 350, 600]),
-                      'mnist-784-euclidean': (150, 32, [100, 200, 350]),
-                      'sift-128-euclidean': (200, 32, [150, 300, 500])}
+    # for every dataset the params are: (is_multi, data_type, ef_construction, M, ef_runtime).
+    dataset_params = {'glove-25-angular': (False, VecSimType_FLOAT32, 100, 16, [50, 100, 200]),
+                      'glove-50-angular': (False, VecSimType_FLOAT32, 150, 24, [100, 200, 300]),
+                      'glove-100-angular': (False, VecSimType_FLOAT32, 250, 36, [150, 300, 500]),
+                      'glove-200-angular': (False, VecSimType_FLOAT32, 350, 48, [200, 350, 600]),
+                      'mnist-784-euclidean': (False, VecSimType_FLOAT32, 150, 32, [100, 200, 350]),
+                      'sift-128-euclidean': (False, VecSimType_FLOAT32, 200, 32, [150, 300, 500])}
```

Comment:
Should we have multi/fp64 datasets here?

Reply:
yes?

Reply:
not necessary for now....